### ISSUE TITLE
paddle中计算entropy过程中没有对概率进行归一化

### ISSUE THREAD

**Issue by darewolf007** [PaddlePaddle/Paddle#56289]
### bug描述 Describe the Bug

```
from paddle.distribution import  Categorical
import paddle
dist = Categorical(logits=[0.5535528,  0.20714243, 0.01162981, 0.51577556, 0.36369765, 0.2609165 ])
print(dist.entropy())
logits = dist.logits - paddle.max(dist.logits, axis=-1, keepdim=True)
print(logits)
```
在库中计算负熵过程中entropy()直接使用了这一部分的logits，没有取对数也没有进行归一化，和计算质量分数的熵计算公式不符
neg_entropy = paddle.sum(prob * (**logits** - paddle.log(z)), axis=-1)
在pytorch中logits在init过程中经过了归一化处理
```
 # Normalize
 self.logits = logits - logits.logsumexp(dim=-1, keepdim=True)
```
